client/components/changed-badge.js

```diff
@@ -1,15 +1,16 @@
 import React from 'react';
+import { connect } from 'react-redux';
 
-class ChangedBadge extends React.Component {
-  render() {
-    if (this.props.fields.some( k=> this.props.latest.some( l=> l.includes(k)))) {
-      return <span className="badge changed">changed</span>;
-    }
-    if (this.props.fields.some( k=> this.props.granted.some( l=> l.includes(k)))) {
-      return <span className="badge">amended</span>;
-    }
-    return null;
+const ChangedBadge = ({ fields, latest, granted }) => {
```

Comment:
I would probably default fields to an empty array here, as it will throw an error if undefined for any reason

Reply:
BUMP